be/src/exec/pipeline/scan/connector_scan_operator.h

```diff
@@ -77,6 +78,17 @@ class ConnectorScanOperator : public ScanOperator {
     ChunkBufferTokenPtr pin_chunk(int num_chunks) override;
     bool is_buffer_full() const override;
     void set_buffer_finished() override;
+
+    int available_pickup_morsel_count() override;
+    void begin_pull_chunk(ChunkPtr res) override;
```

Comment:
these are common scan operator metrics. they can be collected in `ScanOperator` without expensive virtual function call

Reply:
Yeah, I'lll update that in another PR.